web/src/user_profile.ts

```diff
@@ -540,6 +545,67 @@ export function show_user_profile_access_error_modal(): void {
     });
 }
 
+function add_user_to_groups(group_ids: number[], user_id: number, $alert_box: JQuery): void {
+    let success_count = 0;
+
+    // Recursive function to process groups in series (one after another)
+    function add_user_to_next_group(index: number): void {
```

Comment:
Would be better for this function to just accept group ID as parameter. And then to find next group ID can use length of `group_ids_successfully_added` to find index for next group ID.